metrics/config_observability_test.go

```diff
@@ -49,38 +48,47 @@ func TestObservabilityConfiguration(t *testing.T) {
 			"logging.enable-var-log-collection":           "true",
 			"logging.request-log-template":                `{"requestMethod": "{{.Request.Method}}"}`,
 			"logging.revision-url-template":               "https://logging.io",
-			"logging.enable-request-log":                  "false",
+			"logging.enable-request-log":                  "true",
 			"metrics.request-metrics-backend-destination": "stackdriver",
 			"profiling.enable":                            "true",
 		},
 	}, {
 		name:       "observability config with no map",
-		wantErr:    false,
 		wantConfig: defaultConfig(),
 	}, {
-		name:       "invalid request log template",
-		wantErr:    true,
-		wantConfig: nil,
+		name:    "invalid request log template",
+		wantErr: true,
 		data: map[string]string{
 			"logging.request-log-template": `{{ something }}`,
 		},
 	}, {
-		name:    "observability configuration with request log set and with template not set",
+		name:    "observability configuration with request log and template not set",
 		wantErr: false,
 		wantConfig: &ObservabilityConfig{
 			EnableProbeRequestLog:  true,
 			EnableProfiling:        true,
-			EnableRequestLog:       false,
 			EnableVarLogCollection: true,
 			LoggingURLTemplate:     "https://logging.io",
-			RequestLogTemplate:     "",
 			RequestMetricsBackend:  "stackdriver",
 		},
 		data: map[string]string{
 			"logging.enable-probe-request-log":            "true",
+			"logging.enable-request-log":                  "false",
 			"logging.enable-var-log-collection":           "true",
+			"logging.request-log-template":                "",
+			"logging.revision-url-template":               "https://logging.io",
+			"metrics.request-metrics-backend-destination": "stackdriver",
+			"profiling.enable":                            "true",
+		},
+	}, {
+		name:    "observability configuration with request log set and template not set",
```

Comment:
My understanding is that if request log set but template not set, we use default tempalte. If template set to empty, throw an error. Is my understanding wrong?

Reply:
yep that's why it's zeroed out in the test.